tests/System.Buffers.Primitives.Tests/BufferReferenceTesting.cs

```diff
@@ -106,15 +106,12 @@ static void Buffer(OwnedBuffer<byte> buffer)
         static void Pin(OwnedBuffer<byte> buffer)
         {
             var memory = buffer.Buffer;
-            Assert.False(buffer.IsRetained);
```

Comment:
After all the checks to IsRetained are gone, it seems like the tests are not testing anything.